### PR TITLE
 GTX 318 - Containerisation, Make commands, formatter, linter

### PR DESCRIPTION
**Pull request by gwenso**
Added containerisation, Make commands, formatter, linter. .js files were auto/manually tidied up as per formatter and linter.
New PR with merges

### CODE REVIEW

**Review by abindukalpa — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,6 @@ node_modules/
 .DS_Store
 .env.live
```
Comment (cma06) [inline .gitignore]: could we do .env* to encapsulate everything that could come up

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM mcr.microsoft.com/playwright:v1.45.1-jammy
+
+COPY . .
+
+RUN npm install
+RUN npx playwright install
```
Comment (cma06) [inline Dockerfile]: do we have to install playwright like this or can we not have it in the package.json file instead and have `npm install` command pick it up form there?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+EXPOSE 9329
+
+ENTRYPOINT [ "npx", "playwright", "test" ]
```
Comment (cma06) [inline Dockerfile]: This would be better to execute a script command that is defined in the `package.json` file

File: Dockerfile (hunk 2 above)
Comment (gwenso) [reply Dockerfile]: yep you're right, just tested it without and it installed fine so I'll remove that from there and the makefile

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+debug: install
+	npx playwright test --debug
+
+install: $(INSTALL_STAMP)
+
+$(INSTALL_STAMP): package.json package-lock.json
+	npm install
+	npx playwright install
+	npm install --save-dev prettier husky lint-staged
+	npx husky install
```
Comment (cma06) [inline Makefile]: Is this not installed already as it is found in the `package.json` file?

Comment (cma06) [reply Makefile]: Same with playwright

File: Dockerfile (hunk 3 above)
Comment (gwenso) [reply Dockerfile]: any suggestions for that, while keeping it flexible? I agree that it's good practice, but with this way above we can pass the ENV for which environment we want when we run the image (e.g. docker run -e NODE_ENV=live ...)

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	touch $(INSTALL_STAMP)
+
+lint:
```
Comment (cma06) [inline Makefile:28]: Could we make this run via docker only? That way we shouldn't require npm being a required local dependency?

I'm thinking to run this via an image that has npm already and just mount the whole root of the project to it and that way we run the linter inside docker too

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# stop npm installing for every locally run command but still being a dependency before they can be run first
+INSTALL_STAMP = .npm_installed
+# docker build as dependency but only build it the first itme
+IMAGE_STAMP = .image_built
+
+# headless run
+test: install
```
Comment (cma06) [inline Makefile:13]: I was thinking that we remove any commands that would run locally and only focus on running every single command in docker for the purpose of this ticket

File: .gitignore (hunk 1 above)
Comment (gwenso) [reply .gitignore]: made changes in upcoming commit

File: Dockerfile (hunk 2 above)
Comment (gwenso) [reply Dockerfile]: made changes in upcoming commit

File: Dockerfile (hunk 3 above)
Comment (gwenso) [reply Dockerfile]: made changes in upcoming commit

File: Makefile (hunk 4 above)
Comment (gwenso) [reply Makefile]: made changes in upcoming commit

File: Makefile (hunk 5 above)
Comment (gwenso) [reply Makefile:28]: as discussed, for now, I think it's simplest to keep the docker stuff separate as is due to debugging requiring a local install but we can keep this in mind

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+# run with browser open
+debug: install
+	@echo debugging tests in $(ENV) env
+	npx playwright test --debug
+
+install: $(INSTALL_STAMP)
+
+$(INSTALL_STAMP): package.json package-lock.json
+	npm install
+	npx husky install
```
Comment (cma06) [inline Makefile]: This can also be removed

**Pull request merged**